python/graphstorm/graphpartitioning/array_readwriter/numpy_array.py

```diff
@@ -0,0 +1,43 @@
+import logging
+
+import numpy as np
+from numpy.lib.format import open_memmap
+
+from .registry import register_array_parser
+
+
+@register_array_parser("numpy")
+class NumpyArrayParser(object):
+    """ Wrapper class for numpy read/write operations."""
+    def __init__(self):
+        pass
+
+    def read(self, path):
+        """ Read operation on a file.
+
+        Argument
+        --------
+        path: str
+            Path of a file from which to read.
+        """
+        logging.debug("Reading from %s using numpy format" % path)
+        arr = np.load(path, mmap_mode="r")
```

Comment:
what is the numpy file format?

Reply:
does it have to be written by `open_memmap`?

Reply:
Not necessarily. But the expected use of all the ``write`` calls in this directory are meant to be for very small files. In the current implementation, these calls are ONLY used in the end-to-end tests. That is the only use case for these write function calls. 

If there is a need to support these function calls for general purpose usage, then this needs to be investigated further to explore the balance between speed and resource(memory) usage.